### PR TITLE
Separate evaluation logs from the model logs by generating separate log files in a reports directory

### PR DESCRIPTION
**Pull request by Munsio**
@zimmski please check it out if it is working as you like or if there is anything missing.

### CODE REVIEW

**Comment by zimmski** [comment]
Finally, taking this PR!

File: log/logger.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+
+	file, err := os.OpenFile(path, os.O_APPEND|os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	loggerClose = func() {
+		if err := file.Close(); err != nil {
+			panic(err)
```
Comment (bauersimon) [inline log/logger.go:24]: Sure its problematic if the log file cannot be closed, but we should maybe not panic. A log file is not critical.

Same for other panics in this file.

**Review by bauersimon — CHANGES_REQUESTED** [review]

File: evaluate/repository.go [hunk 2]
```diff
@@ -2,28 +2,44 @@ package evaluate
...
+	currentDirectoryPath, err := os.Getwd()
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	repositoryPathRelative, err := filepath.Rel(filepath.Join(currentDirectoryPath), repositoryPath)
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+
+	log, logClose, err := log.File(filepath.Join(resultPath, strings.ReplaceAll(model.ID(), "/", "_"), language.ID(), repositoryPathRelative+".log"))
```
[the remark this answers is not in the document]
Comment (zimmski) [reply evaluate/repository.go:29]: Not sure what you are not a fan about, it is totally ok to have slashes. We even want it.

File: log/logger.go (hunk 1 above)
Comment (zimmski) [reply log/logger.go:24]: When fie closing throws an error, there is something seriously wrong with the system. Don't think we should complicate the API unnessary.

File: evaluate/repository.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bauersimon) [reply evaluate/repository.go:29]: https://www.baeldung.com/linux/filename-with-slash :thinking:

[the remark this answers is not in the document]
Comment (bauersimon) [reply evaluate/repository.go:29]: guess the slashes won't end up in the filename but actually mimic the directory structure of the repository then?

[the remark this answers is not in the document]
Comment (zimmski) [reply evaluate/repository.go:29]: Ah now i get it... yes it is a file path. the join handles that same as when you do

filepath.Join(dir, relativeDir, anotherRelativeDir, fileName)

**Review by bauersimon — APPROVED** [review]

**Pull request merged**